### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ngsp94**
Changes
- CloudyWebAPI added
- Streaming moved into this plugin
- CloudySaveManager developed

Not Changed
- CloudyPanelPlugin's name - will change if the current name causes confusion. let me know.

### CODE REVIEW

**Comment by 9gix** [comment]
1. Shall we merge RemoteController into this repository?
2. Shall we rename this repository into CloudyGamePlugin

**Comment by ngsp94** [comment]
1. Haven't tried out remote controller yet, I have a hunch it doesn't work out-of-the-box for me, may need some tweaks. May need to pull some stuff from CloudyWeb also, like IP and port. So may need to merge.
2. Yes if remote controller is added. No if not. I was referring to the module name, not the plugin name. See, there's already confusion.

File: Source/CloudyWebAPI/Private/CloudyWebAPI.cpp [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+#include "HideWindowsPlatformTypes.h"
+#include "string"
+
+// Credentials of the robot user
+#define USERNAME "joel"
+#define PASSWORD "1234"
+
+DEFINE_LOG_CATEGORY(CloudyWebAPILog);
+
+static const FString BaseUrl = "http://127.0.0.1:8000";
```
Comment (9gix) [inline Source/CloudyWebAPI/Private/CloudyWebAPI.cpp]: use system environment to determine the base url

File: Source/CloudyWebAPI/Private/CloudyWebAPI.cpp [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+* @param PlayerControllerId      Player controller ID of the player who is saving the file
+*
+*/
+void CloudyWebAPIImpl::ReadAndStoreSaveFileURL(FString JsonString, int32 PlayerControllerId)
+{
+    TSharedPtr<FJsonObject> JsonObject = MakeShareable(new FJsonObject());
+    TSharedRef<TJsonReader<TCHAR>> JsonReader = TJsonReaderFactory<TCHAR>::Create(JsonString);
+    FJsonSerializer::Deserialize(JsonReader, JsonObject);
+
+    if (PlayerControllerId == 0)
```
Comment (ngsp94) [inline Source/CloudyWebAPI/Private/CloudyWebAPI.cpp:302]: Can use list of URLs instead? Hard to increase PlayerControllerIds in future.

**Pull request merged**